File: sdkwork-ai-sdk-typescript/src/types/choice.ts
import type { Logprobs } from './logprobs';
import type { Message } from './message';

/** 选择项 */
export interface Choice {
  /** 索引 */
  index?: number;
  /** 消息内容 */
  message?: Message;
  /** 对数概率信息 */
  logprobs?: Logprobs;
  /** 停止生成的原因 */
  finish_reason?: string;
  /** 内容审核命中类型 */
  moderation_hit_type?: string;
}

File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/model/Choice.java
package com.sdkwork.ai.model;

public class Choice {
    private Integer index;
    private Message message;
    private Logprobs logprobs;
    private String finishReason;
    private String moderationHitType;

    public Integer getIndex() {
        return this.index;
    }
    
    public void setIndex(Integer index) {
        this.index = index;
    }

    public Message getMessage() {
        return this.message;
    }
    
    public void setMessage(Message message) {
        this.message = message;
    }

    public Logprobs getLogprobs() {
        return this.logprobs;
    }
    
    public void setLogprobs(Logprobs logprobs) {
        this.logprobs = logprobs;
    }

    public String getFinishReason() {
        return this.finishReason;
    }
    
    public void setFinishReason(String finishReason) {
        this.finishReason = finishReason;
    }

    public String getModerationHitType() {
        return this.moderationHitType;
    }
    
    public void setModerationHitType(String moderationHitType) {
        this.moderationHitType = moderationHitType;
    }
}
